controller/mri/mri.controller.js

```diff
@@ -402,20 +415,18 @@ const apiMriPost = async function (req, res) {
     } else {
       console.log('Start download:');
       downloadQueue[myurl] = { success: 'downloading', cur: 0, len: 1 };
-      downloadMRI(myurl, req, res, (obj) => {
-        console.log("downloadMRI obj:", obj);
-        if (typeof obj.error === "undefined") {
-          console.log('Download succeeded. Insert in DB, remove from queue');
-          obj.success = true;
-          req.db.get('mri').insert(obj);
-          // downloadQueue[myurl] = obj;
-          delete downloadQueue[myurl];
-        } else {
-          console.log('Download failed:', obj);
-          downloadQueue[myurl] = { success: false, error: `${JSON.stringify(obj.error)}` };
-        }
-      });
-
+      const obj = await downloadMRI(myurl, req, res);
```

Comment:
hi @adityaofficial10
I see you replaced the callback for a promise. I'm not sure this will work the same as before, could you test?
The idea of the callback was not to block execution: the download would be triggered, execution would continue, and the callback function checking if the download succeeded or failed would be only called later. With the new code, execution will await until the download is finished, blocking what comes next.
For checking whether this is the case or not, try to download a really big MRI (f. ex, https://braincatalogue.org/data/Thylacine/MRI.nii.gz which is ~90MB). Could you check then if triggering that download blocks the server when you use `await` versus using the callback?